### PR TITLE
add question objects / config bits to the manual

### PR DESCRIPTION
**Pull request by ring-ring-ring**
### Changes
add question oois to the manual with some basic explanation

related to this issue
https://github.com/minvws/nl-kat-coordination/issues/1471

### CODE REVIEW

File: docs/source/introduction/img/dataflowquestionobjects.png
Comment (praseodym) [inline docs/source/introduction/img/dataflowquestionobjects.png:1]: I'm not sure how you made this diagram, but would it be possible to add the source or make the diagram in Mermaid so it's possible to edit it later?

Comment (ring-ring-ring) [reply docs/source/introduction/img/dataflowquestionobjects.png:1]: for sure; its a mermaid diagram I stole from @Donnype but when using restructuredtext it does not render automagically, hence the png.

Comment (ring-ring-ring) [reply docs/source/introduction/img/dataflowquestionobjects.png:1]: https://github.com/minvws/nl-kat-coordination/issues/1471 < Donny kindly pasted them here - I tried to use the main one but failed to make it work with rst.

Comment (praseodym) [reply docs/source/introduction/img/dataflowquestionobjects.png:1]: Maybe we should figure out why Mermaid doesn't work in reStructuredText? It seems to work fine on docs pages written in Markdown, e.g.:

https://github.com/minvws/nl-kat-coordination/blob/59b12f379211a53efb08706ad2e566cc4fe939b1/docs/source/developer_documentation/rocky.md?plain=1#L94-L119

is rendered on https://docs.openkat.nl/developer_documentation/rocky.html#running-a-boefje.

Comment (ammar92) [reply docs/source/introduction/img/dataflowquestionobjects.png:1]: Strange indeed, it should also work without embedding it in Markdown. I've updated `sphinxcontrib-mermaid` (0.8.1 -> 0.9.2) in this branch, perhaps you could try this again @ring-ring-ring?

Comment (ring-ring-ring) [reply docs/source/introduction/img/dataflowquestionobjects.png:1]: image

Computer says no. Any ideas?

Comment (ammar92) [reply docs/source/introduction/img/dataflowquestionobjects.png:1]: I think there's an indentation problem with your example, but also an error in the syntax (`end` shouldn't be there). Anyways I managed to get this documentation built with Sphinx, but it look like `rstcheck` doesn't know the `mermaid` directive which causes the hook to fail. 

image

@Darwinkel any ideas on how to make this work with `rstcheck` as well?

Comment (ammar92) [reply docs/source/introduction/img/dataflowquestionobjects.png:1]: > @Darwinkel any ideas on how to make this work with `rstcheck` as well?

I was thinking if there's no good solution we could [ignore](https://rstcheck.readthedocs.io/en/latest/usage/config/#ignore-directives) the `mermaid` directive and optionally use external mermaid files (possible with the current extension) and use a pre-commit hook to check these mermaid files instead.

Comment (praseodym) [reply docs/source/introduction/img/dataflowquestionobjects.png:1]: For completeness: this `rstcheck` change was done in https://github.com/minvws/nl-kat-coordination/commit/627ebacc222de08a88498e7213800de78358f18b

Comment (ammar92) [reply docs/source/introduction/img/dataflowquestionobjects.png:1]: > For completeness: this `rstcheck` change was done in [627ebac](https://github.com/minvws/nl-kat-coordination/commit/627ebacc222de08a88498e7213800de78358f18b)

@praseodym Thanks for noticing this, I forgot to update this thread

Comment (Darwinkel) [reply docs/source/introduction/img/dataflowquestionobjects.png:1]: @ammar92 I'm not aware of standalone mermaid checkers nor good ways to modify rstcheck. I think using the `ignore` directive is appropriate here.

**Review by Darwinkel — APPROVED** [review]

**Review by praseodym — APPROVED** [review]

**Review by dekkers — APPROVED** [review]

**Pull request merged**